### PR TITLE
Update the GridMask layer to use BaseImageAugmentationLayer

### LINKED ISSUE #190: Migrate GridMask to use BaseImageAugmentationLayer
https://github.com/keras-team/keras-cv/blob/master/keras_cv/layers/preprocessing/grid_mask.py

Example in RandomShear.

### PR DESCRIPTION
**Pull request by qlzh727**
Fix #190

### CODE REVIEW

File: keras_cv/layers/preprocessing/grid_mask.py [hunk 1]
```diff
@@ -141,43 +142,42 @@ def _compute_mask(self, image_height, image_width):
...
+                        int(tf.cast(grid_block, tf.float32) * self.ratio + 0.5), 1
                     ),
-                    gridblock - 1,
+                    grid_block - 1,
                 ),
                 tf.int32,
             )
 
         for _ in range(2):
             start_x = tf.random.uniform(
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/grid_mask.py:171]: should this use _random_generator too?

Comment (LukeWood) [reply keras_cv/layers/preprocessing/grid_mask.py:171]: Also - can we move compute mask to get_random_transformation or were there some sort of vectorization issues requiring it to be in the main func?

**Review by LukeWood — APPROVED** [review]

**Comment by chjort** [comment]
Moving GridMask to BaseImageAugmentationLayer is also being done in PR #159. And the implementation in #159 also makes use of `fill_utils.py`. Auto vectorization is not yet functional, and I am uncertain if it is possible for this layer. It is being worked on though.

Let me know if #159 is still relevant or if i should close it.

**Comment by qlzh727** [comment]
> Moving GridMask to BaseImageAugmentationLayer is also being done in #159. And the implementation in also makes use of `fill_utils.py`. #159 originally proposed a fully vectorized implementation of GridMask, but has now been migrated to use BaseImageAugmentationLayer as per request.
> 
> Auto vectorization however is not yet functional using BaseImageAugmentationLayer, and I am uncertain if it is possible for GridMask since it uses `RandomRotation` layer. It is being worked on though.
> 
> Let me know if #159 is still relevant or if I should close it in light of this PR.

I see. I think your PR is nice since it further cleanup the cl for mask generation. Let me review your PR instead.

**Pull request closed without merging**